src/httpClient.ts

```diff
@@ -51,142 +47,58 @@ export class HttpClient {
   async fetch(
     input: string,
     init?: RequestInit | undefined,
-    maxRetries: number = DEFAULT_MAX_RETRIES,
-    retryDelay: number = DEFAULT_RETRY_DELAY,
   ): Promise<Response> {
     const url = this.baseURL + input;
 
-    let lastError: Error | null = null;
-    let currentRetryDelay = retryDelay;
-
-    for (let attempt = 0; attempt < maxRetries; attempt++) {
-      try {
-        const response = await this.attemptFetch(url, init);
-
-        // Check for OK response
-        if (response.ok) return response;
-
-        // If not ok, then generate an error message
-        const errorMessage = this.generateErrorMessage(response, url);
-        lastError = new Error(errorMessage);
-
-        // Fail fast for non-transient errors
-        if (this.nonTransientError(response)) break;
+    const finalInit = {
+      ...init,
+      headers: {
+        ...init?.headers,
+        "Content-Type": init?.headers?.["Content-Type"] ?? "application/json",
+        authorization: `Bearer ${this.apiKey}`,
+      },
+    };
 
-        // Retry logic for transient errors
-        if (this.shouldRetry(response)) {
-          await this.waitForRetry(response, currentRetryDelay);
-          currentRetryDelay *= 2;
-        } else {
-          // For other errors, throw an error
-          throw new Error(errorMessage);
-        }
-      } catch (error) {
-        lastError = error as Error;
+    const IS_DEBUG = this.debug;
 
-        // Fail fast for a non-transient error
-        if (!this.isNetworkError(error)) break;
-        await this.delay(currentRetryDelay);
-        currentRetryDelay *= 2;
+    if (IS_DEBUG) {
+      console.log("\n");
+      console.debug(`Fetching URL: ${url}`);
+      console.debug(`Method: ${finalInit.method || "GET"}`);
```

Comment:
I'd suggest adding all of these in one `console.debug` line, so that nothing gets logged in between them.
Either create an object and log it:
```
const debugRequest = { url, method: finalInit.method, body: finalInit.body, headers: finalInit.headers };
console.log(`Fetch Request: ${JSON.stringify(debugRequest)}`)
```

or I think you can do this:
```
console.log(
  "Fetching URL", url,
   "Method", method,
 ...)
 
```

Reply:
Done